File: Leetcode/buy_sell_stock/buy_sell_stock.py

class Solution:
    def maxProfit(self, prices):
        maxprofit=0
        lowest=prices[0]
        for i in range(1, len(prices)):
            if lowest > prices[i]:
                lowest = prices[i]
            else:
                currprofit = prices[i] - lowest
                if currprofit  > maxprofit:
                    maxprofit = currprofit
        return maxprofit



if __name__ == "__main__":
    s = Solution()
    print(s.maxProfit([7,11,1,2,4]))

       

File: Leetcode/buy_sell_stock/buy_sell_stock.cpp
#include <vector>
#include <iostream>
using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int lowest = prices[0];
        int maxprofit = 0;
        for (int i = 1; i < prices.size(); i++) {
            lowest = min(lowest, prices[i]);
            maxprofit = max(maxprofit, prices[i] - lowest);
        }
        return maxprofit;
    }
};


int main() {
    Solution* s = new Solution();
    vector<int> v = vector<int>({7,6,4,3,1});
    cout << s->maxProfit(v);
    return 0;
}